client/components/common/tce-core/UploadBtn.vue

```diff
@@ -52,6 +51,12 @@ export default {
     label: { type: String, default: 'Choose a file' },
     sm: { type: Boolean, default: false }
   },
+  methods: {
+    async uploadFile(e) {
```

Comment:
consider renaming to validateAndUpload (since it might be a bit confusing looking at component context having upload and uploadFile methods)